openvdb/openvdb/tree/ValueAccessor.h

```diff
@@ -54,7 +56,13 @@ template<typename TreeType, bool IsSafe = true, Index L0 = 0, Index L1 = 1>
 class ValueAccessor2;
 template<typename TreeType, bool IsSafe = true, Index L0 = 0, Index L1 = 1, Index L2 = 2>
 class ValueAccessor3;
-template<typename TreeCacheT, typename NodeVecT, bool AtRoot> class CacheItem;
+template<typename TreeCacheT, typename NodeVecT, bool AtRoot
+#if OPENVDB_ABI_VERSION_NUMBER >= 10
+, bool BypassLeafAPI =
+        std::is_same<typename NodeVecT::Front, typename NodeVecT::Front::LeafNodeType>::value &&
+        std::is_same<typename NodeVecT::Front::LeafNodeType::Buffer::StorageType, typename NodeVecT::Front::ValueType>::value
```

Comment:
If I'm reading this correctly - this disables this optimization for Mask Grids where the BuildType != ValueType as expected, but don't we also wish to disable this for Bool Grids where BuildType == ValueType but delayed-loading is still disabled, so nothing to gain in using this optimization and it could even be more expensive?

Reply:
LeafBuffer StorageType for bool grids isn't bools, it's the same type as mask grids i.e. `NodeMaskType::Word` - so this is disabled for bool grids too. There's a small static assert test I've added which checks this as well

Reply:
Got it, thanks for correcting me there.